src/power_grid_model_io/data_stores/base_data_store.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model IO project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""
+Abstract data store class
+"""
+
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+import structlog
+
+T = TypeVar("T")
+
+
+class BaseDataStore(Generic[T], ABC):
+    """
+    Abstract data store class
+    """
+
+    def __init__(self):
+        """
+        Initialize a logger
+        """
+        self._log = structlog.get_logger(type(self).__name__)
+
+    @abstractmethod  # pragma: no cover
+    def load(self) -> T:
+        """
+        The method that loads the data from one or more sources and returns it in the specified format.
+        Note that the load() method does not recieve a reference to the data source(s); i.e. the data souce(s)
+        should be set in the constructor, or in a separate member method.
+        """
```

Comment:
an empty method should have a `pass` in the body.

Reply:
Not necessarily; `pass` is just a placeholder to fix the syntax. In this case, the comment serves as a placeholder. I'm not entirely sure, but I guess the liner (pylint) would complain about an extra `pass`.